includes/abstracts/abstract-wc-data.php

```diff
@@ -148,6 +151,26 @@ public function get_data() {
 	}
 
 	/**
+	 * Returns array of expected data keys for this object.
+	 *
+	 * @since 2.7.0
+	 * @return array
+	 */
+	public function get_data_keys() {
+		return array_keys( $this->data );
+	}
+
+	/**
+	 * Returns all "extra" data keys for an object (for sub objects like product types).
+	 *
+	 * @since 2.7.0
+	 * @return array
+	 */
+	public function get_extra_data_keys() {
+		return array_keys( $this->extra_data );
```

Comment:
Are we making 'extra_data' our standard for subclasses? I know we used it in products but wasn't sure if we wanted to do this on all other classes too. Given it's merged with ->data, is this required?